lib/publiccloud/provider.pm

```diff
@@ -230,37 +220,87 @@ sub create_instance {
     $args{instance_type} //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
     $args{image} //= $self->get_image_id();
 
+    my @vms = $self->terraform_apply(count => '1');
+    my $instance = $vms[0];
+    record_info('INFO', "VM created '" . Dumper $instance);
+
+    my $timeout = 60;
+    while ($timeout > 0) {
+        last if script_run("nc -vz -w 1 $instance->{public_ip} 22") == 0;
+        $timeout -= 1;
+        sleep 5;
+    }
+    die('Cannot reach port 22 on IP ' . $instance->{public_ip}) if ($timeout == 0);
+    return $instance;
+}
+
+=head2 terraform_apply
+
+Calls terraform tool and applies the corresponding configuration .tf file
+
+=cut
+sub terraform_apply {
+    my ($self, %args) = @_;
+    my @instances;
+
+    $args{count}                //= '1';
+    $args{instance_type}        //= get_var('PUBLIC_CLOUD_INSTANCE_TYPE');
+    $args{image}                //= $self->get_image_id();
+    $args{ssh_private_key_file} //= '.ssh/id_rsa';
+    $args{user}                 //= $self->username;
```

Comment:
just use `$self->username` don't think that we will ever want to pass this outside

Reply:
shall we keep also $args{user} as @cfconrad mentioned above?

Reply:
I think his comment was explicitly about instance type

Reply:
Is user actually configurable or is it just a fixed value. I only found `admin_username` in azure.tf.

Reply:
normally, it's defined by the provider and used to access the VM with ssh. I would say, it should be always the same, no sense to have different user names here.